test/extended/router/gatewayapicontroller.go

```diff
@@ -88,16 +95,49 @@ var _ = g.Describe("[sig-network-edge][OCPFeatureGate:GatewayAPIController][Feat
 		gatewayClass := buildGatewayClass(gatewayClassName, gatewayClassControllerName)
 		_, err = oc.AdminGatewayApiClient().GatewayV1().GatewayClasses().Create(context.TODO(), gatewayClass, metav1.CreateOptions{})
 		if err != nil && !apierrors.IsAlreadyExists(err) {
-			e2e.Failf("Failed to create GatewayClass %q", gatewayClassName)
+			e2e.Failf("Failed to create GatewayClass %q: %v", gatewayClassName, err)
 		}
 	})
 
 	g.AfterAll(func() {
-		g.By("Cleaning up the GatewayAPI Objects")
+		g.By("Deleting the gateways")
+
 		for _, name := range gateways {
-			err = oc.AdminGatewayApiClient().GatewayV1().Gateways("openshift-ingress").Delete(context.Background(), name, metav1.DeleteOptions{})
+			err = oc.AdminGatewayApiClient().GatewayV1().Gateways(ingressNamespace).Delete(context.Background(), name, metav1.DeleteOptions{})
 			o.Expect(err).NotTo(o.HaveOccurred(), "Gateway %s could not be deleted", name)
 		}
+
+		g.By("Deleting the GatewayClass")
+
+		if err := oc.AdminGatewayApiClient().GatewayV1().GatewayClasses().Delete(context.Background(), gatewayClassName, metav1.DeleteOptions{}); err != nil && !apierrors.IsNotFound(err) {
+			e2e.Failf("Failed to delete GatewayClass %q", gatewayClassName)
+		}
+
+		g.By("Deleting the Istio CR")
+
+		o.Expect(oc.AsAdmin().Run("delete").Args("--ignore-not-found=true", "istio", istioName).Execute()).Should(o.Succeed())
```

Comment:
Istio CR is supposed to be garbage collected since [its owner reference is gatewayclass](https://github.com/openshift/cluster-ingress-operator/blob/be41be00b822b820b03a7190732e3fc983efddf3/pkg/operator/controller/gatewayclass/istio.go#L35-L41).

Reply:
The owner reference on the Istio CR didn't cause it to be deleted (see https://github.com/openshift/origin/pull/29900#issuecomment-2971445559).

Reply:
Comment added in https://github.com/openshift/origin/compare/38d8018dfd320088688bd559b77c7f73e998ef13..e2271274f2b3f34c7eff8ffa252520acb593cacc.